### ISSUE TITLE
Change Loading and Saving

### ISSUE THREAD

**Issue by khanfarhan10** [SiggiGue/hdfdict#4]
#### While Performing :
```
hdfdict.load(fname)
OR
hdfdict.dump(ans, newfname)
```

The following `H5pyDeprecationWarning` occurs :
```
/usr/local/lib/python3.7/dist-packages/hdfdict/hdfdict.py:22: H5pyDeprecationWarning: The default file mode will change to 'r' (read-only) in h5py 3.0. To suppress this warning, pass the mode you need to h5py.File(), or set the global default h5.get_config().default_file_mode, or set the environment variable H5PY_DEFAULT_READONLY=1. Available modes are: 'r', 'r+', 'w', 'w-'/'x', 'a'. See the docs for details.
  yield h5py.File(hdf, *args, **kwargs)
```

A plausible solution  in to pass in the `**args` beforehand for both load and dump (`r` and `w` respectively).

when performing
```
@contextmanager
def hdf_file(hdf, lazy=True, *args, **kwargs):
    """Context manager yields h5 file if hdf is str,
    otherwise just yield hdf as is."""
    if isinstance(hdf, str):
        if not lazy:
            with h5py.File(hdf, *args, **kwargs) as hdf:
                yield hdf
        else:
            yield h5py.File(hdf, *args, **kwargs)
    else:
        yield hdf
```
inside of line _93_ `def load(hdf, lazy=True, unpacker=unpack_dataset, *args, **kwargs):`

```
    with hdf_file(hdf, lazy=lazy, *args, **kwargs) as hdf:
        if lazy:
            data = LazyHdfDict(_h5file=hdf)
        else:
            data = {}
        return _recurse(hdf, data)
```
to 
```
    with hdf_file(hdf, lazy=lazy, 'r') as hdf:
        if lazy:
            data = LazyHdfDict(_h5file=hdf)
        else:
            data = {}
        return _recurse(hdf, data)
```

#### What Should work but doesn't **Right Now** :
```
hdfdict.load(fname,'r')
OR
hdfdict.dump(ans, newfname,'w')
```

###### But again why read and write mode separately (explicitly) mentioned when it can be just implicitly implemented under the hood?

**Comment by khanfarhan10** [comment]
If you wish to, I can work on this possible future bugfix.

**Comment by SiggiGue** [comment]
Solved in v0.4.0.
